src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt

```diff
@@ -0,0 +1,223 @@
+package net.maizegenetics.phgv2.api
+
+import htsjdk.variant.variantcontext.VariantContext
+import htsjdk.variant.vcf.VCFFileReader
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.Channel
+import net.maizegenetics.phgv2.utils.AltHeaderMetaData
+import net.maizegenetics.phgv2.utils.parseALTHeader
+import org.apache.logging.log4j.LogManager
+import java.io.File
+import java.util.*
+
+class HaplotypeGraph(hvcfFiles: List<String>) {
+
+    private val myLogger = LogManager.getLogger(HaplotypeGraph::class.java)
+
+    // Map<sampleName, sampleId>
+    private lateinit var sampleNameToIdMap: Map<String, Int>
+    private lateinit var sampleNames: Array<String>
+
+    // lookup sequence checksum index
+    // lookup[refRangeId][sampleId]
+    private lateinit var lookup: Array<Array<UByte>>
+
+    // seqHash[refRangeId][lookup: UByte]
+    // jagged array because different number of haplotypes for each refRange
+    private lateinit var seqHash: Array<Array<String>>
```

Comment:
Each array entry is an array of sequence hashes for a particular reference range?  How is the reference range associated with this - is there an index mapping that tells you that item 10 maps to a particular reference range hash value?

A more detailed explanation of "lookup" and "seqHash" arrays including how they are interacting would be helpful.

Reply:
Yes, that's refRangeMap